server/src/io/initialhandler.rs

```diff
@@ -205,6 +208,7 @@ fn handle_handshake(ih: &mut InitialHandler, packet: &Handshake) -> Result<(), E
 
 fn handle_request(ih: &mut InitialHandler, packet: &Request) -> Result<(), Error> {
     check_stage(ih, Stage::AwaitRequest, packet.ty())?;
+    let server_icon = (*ih.server_icon).clone().unwrap_or(String::new());
```

Comment:
No need to dereference using `(*ih.server_icon) here—the compiler does that automatically.

Reply:
I'm not too familiar with Arc, but if I try to use:

`ih.server_icon.clone().unwrap_or(String::new())`

I get an `error[E0507]: cannot move out of an Arc` at compile-time. De-referencing explicitly seemed to fix this, unless there is another way?

Reply:
Yep, I guess you're right. The error happens because you're cloning the `Arc` and then unwrapping the `Option`, which consumes it. The alternative would be to use `Option::as_ref()`. but this is fine.